exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -59,6 +59,12 @@
   private final boolean readNumbersAsDouble;
 
   /**
+   * Collection for tracking empty array writers during reading
+   * and storing them for initializing empty arrays
+   */
+  private final Set<ListWriter> emptyArrayWritersSet = Sets.newHashSet();
```

Comment:
Any reason why this needs to be HashSet rather than just a List ?  The HashSet may change the insertion order, so if you had 2 or more empty arrays in the same Json doc within one list,  the output of Flatten could end up changing the order.